src/matter/session/secure/CaseMessages.ts

```diff
@@ -15,56 +23,79 @@ export const KDFSR3_INFO = Buffer.from("Sigma3");
 export const TBE_DATA2_NONCE = Buffer.from("NCASE_Sigma2N");
 export const TBE_DATA3_NONCE = Buffer.from("NCASE_Sigma3N");
 
+/** @see {@link MatterCoreSpecificationV1_0} § 3.5.1 */
+const CRYPTO_GROUP_SIZE_BITS = 256;
+const CRYPTO_GROUP_SIZE_BYTES = 32;
+const CRYPTO_PUBLIC_KEY_SIZE_BYTES = (2 * CRYPTO_GROUP_SIZE_BYTES) + 1;
+
+/** @see {@link MatterCoreSpecificationV1_0} § 3.3 */
+const CRYPTO_HASH_LEN_BYTES = 32;
+const CRYPTO_HASH_BLOCK_LEN_BYTES = 64;
+
+/** @see {@link MatterCoreSpecificationV1_0} § 3.6 */
+const CRYPTO_SYMMETRIC_KEY_LENGTH_BITS = 128;
+const CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES = 16;
+const CRYPTO_AEAD_MIC_LENGTH_BITS = 128;
+const CRYPTO_AEAD_MIC_LENGTH_BYTES = 16;
+const CRYPTO_AEAD_NONCE_LENGTH_BYTES = 13;
+
+/** @see {@link MatterCoreSpecificationV1_0} § 4.13.2.3 */
 export const CaseSigma1T = ObjectT({
-    random: Field(1, ByteStringT()),
-    sessionId: Field(2, UnsignedIntT),
-    destinationId: Field(3, ByteStringT()),
-    ecdhPublicKey: Field(4, ByteStringT()),
-    mrpParams: OptionalField(5, ObjectT({
+    random: Field(1, ByteStringT({ length: 32 })),
+    sessionId: Field(2, BoundedUnsignedIntT({ min: 0, max: 0xFFFF })),
+    destinationId: Field(3, ByteStringT({ length: CRYPTO_HASH_LEN_BYTES })),
+    ecdhPublicKey: Field(4, ByteStringT({ length: CRYPTO_PUBLIC_KEY_SIZE_BYTES })),
+    mrpParams: OptionalField(5, ObjectT({ // sed-parameter-struct
         idleRetransTimeoutMs: OptionalField(1, UnsignedIntT),
         activeRetransTimeoutMs: OptionalField(2, UnsignedIntT),
     })),
-    resumptionId: OptionalField(6, ByteStringT()),
-    resumeMic: OptionalField(7, ByteStringT()),
+    resumptionId: OptionalField(6, ByteStringT({ length: 16 })),
+    resumeMic: OptionalField(7, ByteStringT({ length: CRYPTO_AEAD_MIC_LENGTH_BYTES})),
 });
 
+/** @see {@link MatterCoreSpecificationV1_0} § 4.13.2.3 */
 export const CaseSigma2T = ObjectT({
-    random: Field(1, ByteStringT()),
-    sessionId: Field(2, UnsignedIntT),
-    ecdhPublicKey: Field(3, ByteStringT()),
-    encrypted: Field(4, ByteStringT({ maxLength: 400 })), // TODO: check max length in specs
-    mrpParams: OptionalField(5, ObjectT({
+    random: Field(1, ByteStringT({ length: 32 })),
+    sessionId: Field(2, BoundedUnsignedIntT({ min: 0, max: 0xFFFF })),
+    ecdhPublicKey: Field(3, ByteStringT({ length: CRYPTO_PUBLIC_KEY_SIZE_BYTES })),
+    encrypted: Field(4, ByteStringT()),
+    mrpParams: OptionalField(5, ObjectT({ // sed-parameter-struct
         idleRetransTimeoutMs: OptionalField(1, UnsignedIntT),
         activeRetransTimeoutMs: OptionalField(2, UnsignedIntT),
     })),
 });
 
+/** @see {@link MatterCoreSpecificationV1_0} § 4.13.2.3 */
 export const CaseSigma2ResumeT = ObjectT({
-    resumptionId: Field(1, ByteStringT()),
-    resumeMic: Field(2, ByteStringT()),
-    sessionId: Field(3, UnsignedIntT),
+    resumptionId: Field(1, ByteStringT({ length: 16 })),
```

Comment:
Maybe those are generic enough to be extract as ByteString16T -> this was reduce the minimal memory footprint since the same object will be used everywhere.

Reply:
Ok I can scan over all code tomorow and see ... If then 12 and 32 would be both

Reply:
* length 32 -> 9x in code base
* length 16 -> 5x in code base
* then some more with "constants" that end in 64 or 65 ...

Not sure if it is worth own types :-) What do you think?

But would also see this as a "post PR" topic